lib/active_record/session_store.rb

```diff
@@ -109,8 +111,9 @@ def self.dump(value)
   end
 end
 
-ActiveSupport.on_load(:active_record) do
-  require 'active_record/session_store/session'
+ActiveSupport.on_load(:active_record, yield: true) do
+  next unless ActiveRecord::SessionStore.autoload?(:Session)
+
   ActionDispatch::Session::ActiveRecordStore.session_class = ActiveRecord::SessionStore::Session
```

Comment:
Isn't this duplicated in the session class?

Reply:
Not sure if we should se the session_class when loading session. Do we need to do that?

Reply:
We can't remove the session_class assignment when loading the `session.rb` file and just have the assignment in the lazy hook. This is because of the edge case I mentioned in the PR description.

TL;DR    If the host application reference `Session`, "session.rb" gets loaded, the `active_record` hook is dispatched, at this point the `Session` constant is not yet defined since the AS hook takes precedence, calling `Session` one more time in the hook will just raise a undefined constant error. Chicken and egg situation really

Reply:
Ok, so we should just `require `'active_record/session_store/session'` in this hook them. If the file was already required it will just return `false`.

Reply:
I have a gift for finding overcomplicated solutions 🤦‍♂️ . Thanks!